boards/arm/gd32f450i_eval/gd32f450i_eval_defconfig

```diff
@@ -9,6 +9,8 @@ CONFIG_ARM_MPU=y
 CONFIG_HW_STACK_PROTECTION=y
 CONFIG_CORTEX_M_SYSTICK=y
 
+CONFIG_PINCTRL=y
+
```

Comment:
As far I understand there is no practical application without use at least 1 gpio pin. This means that PINCTRL must be always enabled. My suggestion is enable at HAS_GD32_HAL.

```diff
config HAS_GD32_HAL
	bool
	select HAS_CMSIS_CORE if SOC_FAMILY_GD32_ARM
+	select PINCTRL
```

Reply:
I have added 
```
config PINCTRL
	default y
```
soc/arm/gigadevice/Kconfig.defconfig, it's the usual way to do this. The hal and pinctrl are not related, other than pinctrl drivers use the hal.